src/vllm_router/prefix/hashtrie.py

```diff
@@ -14,17 +14,30 @@
 
 import asyncio
 import logging
-from typing import Generator, Set, Tuple
+import os
+from collections import OrderedDict
+from typing import Dict, Generator, Optional, Set, Tuple
 
+import psutil
 import xxhash
 
+from .config import HashTrieConfig
+
 logger = logging.getLogger(__name__)
 
 
+class NodeMetadata:
+    """Metadata for nodes tracked in LRU cache"""
```

Comment:
You need to make NodeMetadata to be immutable for nodes tracked in LRU cache. Please add `@dataclass(frozen=True)`